### ISSUE TITLE
How to get links to all calendar/tag pages

### ISSUE THREAD

**Issue by aardvarkk** [middleman/middleman-blog#374]
## Expected behaviour

- I'd like to display links to all other pages on the calendar/tag pages within a given calendar/tag page.

## Actual behaviour

- I'm only able to obtain `prev_page` and `next_page` links, and can't figure out a clean interface by which to obtain links to all of the other pages.

## Additional information

- Ruby version: 2.3.8
- Middleman version: 4.3.4
- Middleman Blog version: 4.0.3
- OS version: macOS 10.14.5

**Comment by tdreyno** [comment]
Looks like you are right, that data for the full set is never captured or shared with templates:

It kind of happens here:
https://github.com/middleman/middleman-blog/blob/master/lib/middleman-blog/paginator.rb#L51-L64

Where would it make sense to share that data? Along side prev_page and next_page variables? I can refactor things to make that possible.

**Comment by aardvarkk** [comment]
I started to look into making a variable called `page_list` that would just contain an array of all of the resources. I did find exactly that section of the code and that's where I started making changes. 

What I managed to do was to hack together a helper that would keep wandering through `prev_page` and `next_page` until there were none left. So I would keep accessing `current_resource.locals['prev_page']` and building up an array as I walked through. It was pretty hacky but it worked without me needing to change any code in `middleman-blog`. 

If this is a pretty fringe request I don't mind if you close it off given that there is a workaround -- it just took me quite a while to try to figure out how I could get a list of all the pages and I felt like other people might also want to do that.

**Comment by tdreyno** [comment]
That's clever. Can you put your solution code here for future folks searching for this?

**Comment by aardvarkk** [comment]
Yep -- I should be able to post something tonight. I'll edit this comment with the general idea.

**Comment by markets** [comment]
ℹ️ This issue is stale because it has been open for more than 90 days with no activity. It will be closed in 30 days if no further activity occurs. Thank you for your contributions.
